lib/Reader/readerir.cpp

```diff
@@ -4099,18 +4105,24 @@ bool GenIR::canExpandMDArrayRef(CORINFO_SIG_INFO *Sig, bool IsStore,
   CORINFO_CLASS_HANDLE Class = nullptr;
   uint32_t ElemSize = 0;
   if (IsStore) {
-    CORINFO_ARG_LIST_HANDLE ArgLst = Sig->args;
+    CORINFO_ARG_LIST_HANDLE ArgList = Sig->args;
+    CORINFO_CLASS_HANDLE ArgType;
+
     // Skip arguments for each dimension.
     for (uint32_t R = 0; R < *Rank; R++) {
-      ArgLst = getArgNext(ArgLst);
+      CorType = strip(getArgType(Sig, ArgList, &ArgType));
+      if (CorType != CorInfoType::CORINFO_TYPE_INT) {
```

Comment:
I don't see such cases, but this whole codepath is really just optimization. I think that falling back to the helper in corner cases is okay. If you feel differently, however--or if you'd like the assert simply to help find the corner cases--I can make this change.